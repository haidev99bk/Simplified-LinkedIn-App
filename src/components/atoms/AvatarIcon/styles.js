import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  avatar: {
    height: 24,
    width: 24,
  },
});

export default styles;
