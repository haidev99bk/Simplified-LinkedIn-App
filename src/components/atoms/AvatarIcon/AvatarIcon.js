import {TouchableWithoutFeedback, Image} from 'react-native';
import React from 'react';
import styles from './styles';

export default function AvatarIcon(props) {
  const {source, isPressable, onPress, size, rounded = true, style} = props;

  const handlePress = () => {
    if (!isPressable) {
      return;
    }
    if (onPress) {
      onPress;
    }
  };

  return (
    <TouchableWithoutFeedback onPress={handlePress}>
      <Image
        source={source}
        style={[
          styles.avatar,
          size && {width: size, height: size},
          rounded && {borderRadius: 1000},
          style,
        ]}
      />
    </TouchableWithoutFeedback>
  );
}
