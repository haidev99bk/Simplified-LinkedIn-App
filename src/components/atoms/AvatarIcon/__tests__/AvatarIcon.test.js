import 'react-native';
import React from 'react';
import AvatarIcon from '../AvatarIcon';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('AvatarIcon rendered correctly', () => {
  const snap = renderer.create(<AvatarIcon />).toJSON();

  expect(snap).toMatchSnapshot();
});
