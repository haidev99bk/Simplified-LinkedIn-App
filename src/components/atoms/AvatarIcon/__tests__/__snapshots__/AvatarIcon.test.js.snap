// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarIcon rendered correctly 1`] = `
<Image
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "height": 24,
        "width": 24,
      },
      undefined,
      {
        "borderRadius": 1000,
      },
      undefined,
    ]
  }
/>
`;
